lib/evmone/instructions.cpp

```diff
@@ -491,20 +491,38 @@ const instr_info* op_sstore(const instr_info* instr, execution_state& state) noe
     if (state.msg->flags & EVMC_STATIC)
         return state.exit(EVMC_STATIC_MODE_VIOLATION);
 
+    if (state.rev >= EVMC_ISTANBUL)
+    {
+        const auto correction = state.current_block_cost - instr->arg.p.number;
+        const auto gas_left = static_cast<uint64_t>(state.gas_left + correction);
```

Comment:
This doesn't seem to work right: for `PUSH1 0 PUSH1 0 SSTORE` with 2306 starting gas, at the beginning of SSTORE `state.gas_left` is 2300 and `correction` is 6

Reply:
got it, `arg.p.number` should be filled for SSTORE in analysis